backends/qualcomm/scripts/download_qnn_sdk.py

```diff
@@ -0,0 +1,350 @@
+            )
+            continue
+        shutil.copy(src_path, target_dir / fname)
+
+    print(f"[libcxx] Staged libc++ to {target_dir}")
+
+
+REQUIRED_QNN_LIBS: List[str] = [
+    "libQnnHtp.so",
+]
+
+
+def _ld_library_paths() -> List[pathlib.Path]:
+    """Split LD_LIBRARY_PATH into ordered directories (skip empties)."""
+    raw = os.environ.get("LD_LIBRARY_PATH", "")
+    return [pathlib.Path(p) for p in raw.split(":") if p.strip()]
+
+
+def _find_lib_in_ld_paths(
+    libname: str, ld_dirs: Optional[List[pathlib.Path]] = None
+) -> Optional[pathlib.Path]:
+    """Return first matching path to `libname` in LD_LIBRARY_PATH, or None."""
+    if ld_dirs is None:
+        ld_dirs = _ld_library_paths()
+    for d in ld_dirs:
+        candidate = d / libname
+        try:
+            if candidate.exists():
+                return candidate.resolve()
+        except Exception:
+            # Ignore unreadable / permission issues, keep looking.
+            pass
+    return None
+
+
+def _check_libs_in_ld(
+    libnames: List[str],
+) -> Tuple[bool, Dict[str, Optional[pathlib.Path]]]:
+    """
+    Check if each lib in `libnames` exists in LD_LIBRARY_PATH directories.
+
+    Returns:
+        all_present: True iff every lib was found
+        locations:   mapping lib -> path (or None if missing)
+    """
+    ld_dirs = _ld_library_paths()
+    locations: Dict[str, Optional[pathlib.Path]] = {}
+    for lib in libnames:
+        locations[lib] = _find_lib_in_ld_paths(lib, ld_dirs)
+    all_present = all(locations[lib] is not None for lib in libnames)
+    return all_present, locations
+
+
+# -----------------------
+# Ensure QNN SDK library
+# -----------------------
+def _ensure_qnn_sdk_lib() -> bool:
+    """
+    Ensure libQnnHtp.so is available.
+      - If found in LD_LIBRARY_PATH: do nothing, return True.
+      - Otherwise: ensure packaged SDK is present, then load libQnnHtp.so from it.
+    """
+    all_present, locs = _check_libs_in_ld(REQUIRED_QNN_LIBS)
+    if all_present:
+        print("[QNN] libQnnHtp.so found in LD_LIBRARY_PATH; skipping SDK install.")
+        for lib, p in locs.items():
+            print(f"      - {lib}: {p}")
+        return True
+
+    # Not found → use packaged SDK
+    qnn_sdk_dir = SDK_DIR
+    print(f"[QNN] libQnnHtp.so not found in LD_LIBRARY_PATH.")
+    if not qnn_sdk_dir.exists():
+        print("[QNN] SDK dir missing; downloading...")
+        _download_qnn_sdk()
+    else:
+        print(f"[QNN] Using existing SDK at {qnn_sdk_dir}")
+
+    os.environ["QNN_SDK_ROOT"] = str(qnn_sdk_dir)
+
+    qnn_lib = qnn_sdk_dir / "lib" / "x86_64-linux-clang" / "libQnnHtp.so"
+    print(f"[QNN] Loading {qnn_lib}")
+    try:
+        ctypes.CDLL(str(qnn_lib), mode=ctypes.RTLD_GLOBAL)
+        print("[QNN] Loaded libQnnHtp.so from packaged SDK.")
+        return True
+    except OSError as e:
+        print(f"[QNN][ERROR] Failed to load {qnn_lib}: {e}")
+        return False
```

Comment:
returning from try catch block is a bit strange. Can you just use a boolean and return that

Reply:
> Can you just use a boolean and return that

What does it mean?

Reply:
write it as
```
lib_loaded = False
    try:
        ctypes.CDLL(str(qnn_lib), mode=ctypes.RTLD_GLOBAL)
        print("[QNN] Loaded libQnnHtp.so from packaged SDK.")
        lib_loaded = True
    except OSError as e:
        print(f"[QNN][ERROR] Failed to load {qnn_lib}: {e}")
return lib_loaded
```